src/modules/Game.class.js

```diff
@@ -20,25 +20,93 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
     // eslint-disable-next-line no-console
-    console.log(initialState);
+    this.state = 'idle';
+    this.score = 0;
+    this.initialState = initialState;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveLeft() {
+    if (this.state !== 'idle') {
+      const moveTheLeft = this.initialState.map((row) => mergeToTheLeft(row));
+
+      this.state = loseWinState(moveTheLeft);
+
+      this.score += calculateScore(this.initialState, moveTheLeft);
+
+      const newCellAdded = addNewCellRandomCell(moveTheLeft, 1);
+
+      this.initialState = newCellAdded;
+
+      return this.initialState;
+    }
+  }
+  moveRight() {
+    if (this.state !== 'idle') {
+      const moveTheRight = this.initialState.map((row) => mergeToTheRight(row));
+
+      this.state = loseWinState(moveTheRight);
+
+      this.score += calculateScore(this.initialState, moveTheRight);
+
+      const newCellAdded = addNewCellRandomCell(moveTheRight, 1);
+
+      this.initialState = newCellAdded;
+
+      return this.initialState;
+    }
+  }
+  moveUp() {
+    if (this.state !== 'idle') {
```

Comment:
The condition `this.state !== 'idle'` is incorrect. The game should only allow moves when the state is 'playing'. Change the condition to `this.state === 'playing'`.